### PR TITLE
Emulate A-Frame cursor events in Altspace

### PR DESCRIPTION
**Pull request by amberroy**
- listen for Altspace cursor events, and fire new events with names used by A-Frame: mouseup, mousedown, mouseenter, mouseleave, and click
- added example: test-cursor-events

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -4,7 +4,7 @@ Component to make an [A-Frame](https://aframe.io) scene compatible with [Altspac
 
 Live examples: http://altspacevr.github.io/aframe/examples/ (forked from [aframevr/aframe](https://github.com/aframevr/aframe))
 
-Note that when running in Altspace, the scene will not respond to cursor events nor be synchronized between users.  In addition, the following A-Frame features are currently not supported in Altspace: lights, transparency, and video.  For details, see the [Three.js Feature Support](http://github.com/AltspaceVR/AltspaceSDK#threejs-feature-support) section in the [AltspaceSDK](http://github.com/AltspaceVR/AltspaceSDK) repo. 
+Note that when running in Altspace, the scene will not be synchronized between users.  In addition, the following A-Frame features are currently not supported in Altspace: lights, transparency, and video.  For details, see the [Three.js Feature Support](http://github.com/AltspaceVR/AltspaceSDK#threejs-feature-support) section in the [AltspaceSDK](http://github.com/AltspaceVR/AltspaceSDK) repo.
```
Comment (brianpeiris) [inline README.md]: Oh, I dunno how this happened but it looks like we lost the edits I made about transparency (which is now supported) https://github.com/AltspaceVR/aframe-altspace-component/commit/22a1679a2ae4948d225210224e03f6591f59d4d1

File: dist/aframe-altspace-component.js [hunk 2]
```diff
@@ -120,6 +106,86 @@
...
+	  /********** Helper Methods **********/
+
+	  /**
+	   * Swap in Altspace renderer when running in AltspaceVR.
+	   */
+	  initRenderer: function () {
+
+	    var scene = this.el.object3D;
+	    if (!this.data.usePixelScale) {
+	      console.log('useMeterScale');//XXX
```
Comment (brianpeiris) [inline dist/aframe-altspace-component.js]: Remove?

File: dist/aframe-altspace-component.js [hunk 3]
```diff
@@ -120,6 +106,86 @@
...
+	   * Emulate A-Frame cursor events when running in altspaceVR.
+	   */
+	  initCursorEvents: function() {
+
+	    var scene = this.el.object3D;
+	    var cursorEl = document.querySelector('a-cursor') || document.querySelector('a-entity[cursor]');
+	    if (cursorEl) { 
+	      // Hide A-Frame cursor mesh.
+	      cursorEl.setAttribute('material', 'transparent', true);
+	      cursorEl.setAttribute('material', 'opacity', 0.0);
```
Comment (brianpeiris) [inline dist/aframe-altspace-component.js:151]: Would be better to set `material.visible = false` if possible. Turning on transparency uses a more expensive shader.

**Comment by brianpeiris** [comment]
:+1: Please also update the aframe examples in the gh-pages branch: https://github.com/AltspaceVR/aframe/tree/gh-pages

File: index.js [hunk 4]
```diff
@@ -74,4 +60,84 @@ AFRAME.registerComponent('altspace', {
...
+  /********** Helper Methods **********/
+
+  /**
+   * Swap in Altspace renderer when running in AltspaceVR.
+   */
+  initRenderer: function () {
+
+    var scene = this.el.object3D;
+    if (!this.data.usePixelScale) {
+      console.log('useMeterScale');//XXX
```
Comment (GavanWilhite) [inline index.js]: clear?

File: README.md (hunk 1 above)
Comment (amberroy) [inline README.md]: Good catch, I added it back in.

**Pull request merged**